Engine/system/api/x11api.cpp

```diff
@@ -188,7 +189,21 @@ SystemApi::Window *X11Api::implCreateWindow(Tempest::Window *owner, uint32_t w,
   }
 
 SystemApi::Window *X11Api::implCreateWindow(Tempest::Window *owner, SystemApi::ShowMode sm) {
-  return implCreateWindow(owner,800,600); //TODO
+  Screen* s = DefaultScreenOfDisplay(dpy);
+  int width = s->width;
+  int height = s->height;
+  
+  SystemApi::Window* hwnd = nullptr;
+
+  if(sm==Normal) 
+    hwnd = implCreateWindow(owner,800, 600);
+  else 
+    hwnd = implCreateWindow(owner,width, height);
+  
+  if(sm==FullScreen)
+    setAsFullscreen(hwnd,true);
```

Comment:
While is ok to call `setAsFullscreen`, better alternative is `implSetAsFullscreen`, since `setAsFullscreen` is just a wrapper

Reply:
That has been changed as a result of the switch statement example